### ISSUE TITLE
Mitmproxy not using up the installed certificates

### ISSUE THREAD

**Issue by illera88** [mitmproxy/mitmproxy#4335]
#### Problem Description
It seems that Mitmproxy is not using the certs it created.

#### Steps to reproduce the behavior:
1. `pip install mitmproxy`
2. `mitmdump.exe` // Justo to have mitmproxy generating the certs. Stop it when run
3. `certutil -addstore root "%USERPROFILE%\.mitmproxy\mitmproxy-ca-cert.cer"` // From Admin cmd
4. `mitmdump -p 8081`

From other cmd:
```
curl -x http://127.0.0.1:8081 http://example.com/ // Works
curl -x http://127.0.0.1:8081 https://example.com/  // Errors
curl -x http://127.0.0.1:8081 https://example.com/ -k  // Works
```

Error:
`<< Cannot establish TLS with client (sni: example.com): TlsException('SSL handshake error: ZeroReturnError()')`

#### System Information
```
Mitmproxy: 5.3.0
Python:    3.7.1
OpenSSL:   OpenSSL 1.1.1h  22 Sep 2020
Platform:  Windows-10-10.0.20241-SP0
```

**Comment by Prinzhorn** [comment]
> It seems that Mitmproxy is not using the certs it created.

To me it seems that _curl_ doesn't trust the CA that mitmproxy is using. I might be wrong and I don't have Windows at hand, but it seems curl doesn't use the system CA store on Windows? What happens if you use a browser to request the same URL? Does it work? Does it show the mitmproxy cert?

https://stackoverflow.com/questions/37551409/configure-curl-to-use-default-system-cert-store-on-windows
https://serverfault.com/questions/633644/adding-a-self-signed-cert-to-the-trusted-certs-within-curl-in-windows

**Comment by Prinzhorn** [comment]
Honestly I'm confused if curl does support it now or not (https://github.com/curl/curl/pull/5657)
Anyway, I'm closing this as I think this is an issue with curl, not mitmproxy. Feel free to reopen if you disagree.

**Comment by illera88** [comment]
You are right. Curl is not picking up the certs from the certstore for the version of curl that I have. Just in the case that someone comes along curl 7.71.0 will pick certs from the cert store:
`Starting with libcurl 7.71.0, due to ship on June 24, 2020, it will get the ability to use the Windows CA cert store when built to use OpenSSL`

**Comment by mhils** [comment]
TL;DR for folks dropping by from Google search:

1. Some versions of `curl` do not pick up the certs installed into the Windows certificate store.
2. Workarounds:
  a) `curl --cacert ~/.mitmproxy/mitmproxy-ca-cert.pem ...`
  b) `curl --insecure ...` may be good enough if mitmproxy is running on the same machine
  c) Update `curl`
